packages/dds/map/src/interfaces.ts

```diff
@@ -150,9 +150,10 @@ export interface IDirectory extends Map<string, any>, IEventProvider<IDirectoryE
     set<T = any>(key: string, value: T): this;
 
     /**
-     * Creates an IDirectory child of this IDirectory.
+     * Creates an IDirectory child of this IDirectory, or retrieves the existing IDirectory child if one with the
```

Comment:
createSubDirectory, getSubDirectory, hasSubDirectory - it's enough to have 2 APIs here.
Either create & get, or create & has would be sufficient.